__tests__/app.test.js

```diff
@@ -0,0 +1,40 @@
+const request = require('supertest');
+const seed = require('../db/seeds/seed.js');
+const data = require("../db/data/test-data");
+const db = require('../db/connection.js');
+const app = require('../app.js');
+
+beforeEach(() => {
+    return seed(data);
+});
+afterAll(() => {
+    return db.end();
+});
+
+test("returns 404 bad request when sent invalid endpoint", () => {
```

Comment:
As you're not sending back an error from the models using a ```Promise.reject```, I think the 404s you're receiving are likely to the default ones coming from express, and not from the nice custom error handler you made in ```app.js```. This is completely fine for this ticket - but we don't have to test for express default behaviours, so either change your models so the error is passed to your custom error handler, or just let express deal with it and delete the test and custom error handler until you need it. 🙂 

To check which one it is, try sending a different message back to your custom error handler and testing that.